DXgovernance/DAVI
# Feat: update content hash ens subdomains, add validations and refactor ENS content editor form

# Description

- This enables editing the content of the subdomain. Mostly added validations. The core functionality is left untouched.
- The ENS content editor now uses `react-hook-form` that shows validation errors below the fields and prevents the user from submitting invalid data.
- Added fix for ENS action showing as "Unknown" 

To test it:

1. Make sure you're on Goerli network
2. Make sure the `0xC5B20AdE9c9Cd5e0CC087C62b26B815A4bc1881` account is selected, or an account that is a member of the guild
3. Go to the Swapr guild 
4. Make a new proposal with an action to change the ENS content. For now, the only ENS name is "swpr.eth" and the subdomain "test.swpr.eth"
5. Set the domain or subdomain (excluding the "eth" at the end)
6. Set the IPFS hash
7. Save and create the proposal
8. Vote and execute it
9. Go to https://app.ens.domains/name/swpr.eth/details
10. There should be an IPFS address (like ipfs://bafybeidovn3pkjyhtjiu5c2djp4v5ttsbh56jasiei3by5dgt7lduch6m4 )
11. Clicking on it should display the content set
12. To check the subdomain, click on "Subdomains", at the right of "Details"


![image](https://user-images.githubusercontent.com/75996796/190034036-87db6e6b-9140-4f2a-8c6f-0541cacfcf21.png)


Closes #177 

For future issue: support IPFS CID v1

## Type of change


# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce.

# Checklist:
